client/shaders/nodes_shader/opengl_vertex.glsl

```diff
@@ -241,7 +241,7 @@ void main(void)
 		float pFactor = getPerspectiveFactor(getRelativePosition(m_ShadowViewProj * mWorld * shadow_pos));
 		if (f_normal_length > 0.0) {
 			nNormal = normalize(vNormal);
-			cosLight = dot(nNormal, -v_LightDirection);
+			cosLight = max(1e-5, dot(nNormal, -v_LightDirection));
```

Comment:
why not `max(0, ...)`?

Reply:
We divide by this value later in the code, it would cause positive infinity result and instability.